src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java

```diff
@@ -127,26 +126,20 @@ private static int getLvIndex(String desc, boolean isStatic, int asmIndex) {
 
 		@Override
 		public void visitParameter(String name, int access) {
-			name = ((AsmRemapper) remapper).mapMethodArg(this.owner, this.name, this.desc, getLvIndex(this.desc, isStatic, argsVisited), name);
+			final int lvIndex = getLvIndex(this.desc, isStatic, argsVisited);
+			name = ((AsmRemapper) remapper).mapMethodArg(this.owner, this.name, this.desc, lvIndex, name);
 
 			if (renameInvalidLocals && !isValidJavaIdentifier(name)) {
-				if (this.isAbstract) {
-					Type type = Type.getArgumentTypes(this.desc)[argsVisited];
-					name = getGeneratedName(type);
-				} else {
-					name = renamedInvalidLocals.getOrDefault(argsVisited, name);
-				}
+				final Type type = Type.getArgumentTypes(this.desc)[argsVisited];
+				name = getNameFromType(type, lvIndex);
 			}
 
-			if(!renameInvalidLocals || isValidJavaIdentifier(name)) {
-				argsVisited++;
-				super.visitParameter(name, access);
-			}
+			argsVisited++;
+			super.visitParameter(name, access);
```

Comment:
This will allow `null` parameter names if `name` is null (which as we've seen before end up as `paramx` or `<anonymous>`). Might want to null check `name` but still increase `argsVisited` regardless given `renameInvalidLocals` will be false to get into the situation.

Reply:
Yes, that is by design, since we might still have mappings for parameters down the line. I don't think it really hurts to produce something invalid when the option is not set, as there won't be any change in the LVT as well.

Not doing the call when there's a null value is definitely wrong, since the parameters get added in the order they are encountered.
This would mean, that if we have no mapping for the first parameter, the second parameter's mapping might be applied for the first parameter.

Reply:
I was thinking that the LVT might have a name that could be used instead, but I suppose if you've got `renameInvalidLocals` false consistency between the two is probably not a problem given the lack of concern about the LVT being valid to start with.

Reply:
One thing I could do in `checkParameters` is to first iterate all mappings and see if there is at least one, and only if I find a parameter mapping, actually call `visitParameter`.
That way no mapping happens when we have absolutely none.

Reply:
That's a good middle ground solution for the situation, it shouldn't slow down remapping too much given the maximum number of parameters is only 255, and it's only when `renameInvalidLocals` is false.

Reply:
As Player pointed out on Discord, we need to also look for a valid LVT parameter first.
So I will need to hold the entire parameter generation back to after all LVT names are processed.

Because of this, I'll probably move the mapping to visitLocalVariable entirely and at the end just take whatever this method used.

This will also require to save the access modifier again.

Reply:
In what I just pushed, we get this without renaming invalids:
```
    MethodParameters:
      Name  Flags  
      ----  -----
      \u2603     
      \u2603     
      \u2603     
      \u2603     
    LocalVariableTable:
      Start  Length  Slot  Name  Signature
      -----  ------  ----  ----  --------------------------
      0      26      0     this  Lnet/minecraft/class_4371;
      0      26      1     \u2603     I
      0      26      2     \u2603     I
      0      26      3     \u2603     I
      0      26      4     \u2603     I
```

And in an abstract method without renaming invalids:
```
  protected abstract void method_21112(int p0, int p1, boolean p2);
    Flags: PROTECTED, ABSTRACT
    MethodParameters:
      Name  Flags  
      ----  -----
      null  
      null  
      null  
```

Of course parameters that actually do have a mapping are not affected:
```
    Signature:
      <T:Ljava/lang/Object;>(Lnet/minecraft/util/registry/Registry<TT;>;Ljava/util/function/Function<TT;Ljava/lang/String;>;Ljava/util/Set<Ljava/lang/String;>;)V
    MethodParameters:
      Name             Flags  
      ---------------  -----
      registry         
      keyExtractor     
      translationKeys  
    LocalVariableTable:
      Start  Length  Slot  Name             Signature
      -----  ------  ----  ---------------  --------------------------------------
      0      22      0     registry         Lnet/minecraft/util/registry/Registry;
      0      22      1     keyExtractor     Ljava/util/function/Function;
      0      22      2     translationKeys  Ljava/util/Set;
      4      18      3     \u2603                Lnet/minecraft/util/Language;
```